server/vulnerabilities/nvd/cpe.go

```diff
@@ -217,44 +217,53 @@ func CPEFromSoftware(db *sqlx.DB, software *fleet.Software, translations CPETran
 			return "", fmt.Errorf("getting cpes for: %s: %w", software.Name, err)
 		}
 
+		var match *IndexedCPEItem
 		for _, item := range results {
-			if !item.Deprecated {
-				hasAllTerms := true
+			hasAllTerms := true
 
-				sName := strings.ToLower(software.Name)
-				for _, sN := range strings.Split(item.Product, "_") {
-					hasAllTerms = hasAllTerms && strings.Index(sName, sN) != -1
-				}
+			sName := strings.ToLower(software.Name)
+			for _, sN := range strings.Split(item.Product, "_") {
+				hasAllTerms = hasAllTerms && strings.Index(sName, sN) != -1
+			}
 
-				sVendor := strings.ToLower(software.Vendor)
-				sBundle := strings.ToLower(software.BundleIdentifier)
-				for _, sV := range strings.Split(item.Vendor, "_") {
-					if sVendor != "" {
-						hasAllTerms = hasAllTerms && strings.Index(sVendor, sV) != -1
-					}
+			sVendor := strings.ToLower(software.Vendor)
+			sBundle := strings.ToLower(software.BundleIdentifier)
+			for _, sV := range strings.Split(item.Vendor, "_") {
+				// If we can't check the vendor, then assume is a bad match, this is to avoid any
+				// false positives.
```

Comment:
Hm, but couldn't we match with vendor `*` in this case? What was matching in this case?